docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+A form is like a collection of menus or raw responses and it is basically collecting information from the user.
+
+It is composed by a **header**, a **body** and a **footer**. The body represents form items and these can be slightly modified menus or raw responses.
+
+The form items are being processed in a successive manner. After all the form items have been processed, the user can confirm his choices or reset any of them.
+
+## Json Structure
+
+Your server must return a JSON response with a structure similar to the one below:
+
+```
+{'body': [{'type': 'string',
+           'name': 'descr',
+           'description': 'Provide a description',
+           'method': 'POST',
+           'path': '/callback-path'},
+          {'type': 'date'
+           'name': 'due_date',
+           'header': 'due date',
+           'description': 'Provide a due date'},
+          {'type': 'menu',
+           'name': 'prio',
+           'header': 'priority',
+           'body': [{'type': 'option',
```

Comment:
From these two body items I don't understand how to define a separator item. The `type` would be `content` instead of `option`?

Reply:
That's correct

Reply:
Ok, so `value` becomes optional, right?

Reply:
Yes. It is like the callback path for options in a menu, becomes irrelevant when type is content